frontend/catalyst/compilation_pipelines.py

```diff
@@ -660,38 +208,26 @@ def compile(self):
         if self.compiled_function and self.compiled_function.shared_object:
             self.compiled_function.shared_object.close()
 
-        if self.compiling_from_textual_ir:
-            # Module name can be anything.
-            module_name = "catalyst_module"
-            shared_object, llvm_ir, inferred_func_data = self.compiler.run_from_ir(
-                self.user_function, module_name, self.workspace
-            )
-            qfunc_name = inferred_func_data[0]
-            # Parse back the return types given as a semicolon-separated string
-            with ir.Context():
-                restype = [ir.RankedTensorType.parse(rt) for rt in inferred_func_data[1].split(",")]
-        else:
+        # WARNING: assumption is that the first function
```

Comment:
Is it something that we want to revisit soon?

Reply:
I did add a todo for it, I think yes it would be good to revisit 😅